### PR TITLE
Fix: Self DM message count

### PR DESCRIPTION
**Pull request by mjlamb**
For self DM message the counts where not correct as the LastReadL column date was not being set.
In SetLastRead, we test for self DM and set the LastReadL fixing the message count.

### CODE REVIEW

File: Nostrid.Core/Data/DmService.cs [hunk 1]
```diff
@@ -89,7 +89,19 @@ public void SetLastRead(string accountId, string otherAccountId)
...
+        if (accountId == otherAccountId)
+        {
+            // you can dm yourself, this fixes the message counts.
+            db.DmPairs
+                .Where(p => p.AccountL == accountId && p.AccountH == otherAccountId)
+                .ExecuteUpdate(p => p.SetProperty(p => p.LastReadL, now));
+
+            db.DmPairs
+                .Where(p => p.AccountH == accountId && p.AccountL == otherAccountId)
+                .ExecuteUpdate(p => p.SetProperty(p => p.LastReadH, now));
```
Comment (lapulpeta) [inline Nostrid.Core/Data/DmService.cs:102]: I think these two statements can be combined into one. Can you give it a try?

Comment (mjlamb) [reply Nostrid.Core/Data/DmService.cs:102]: Yeah was thinking that myself, will take a look.

Comment (mjlamb) [reply Nostrid.Core/Data/DmService.cs:102]: Do we even need to report self messages as unread, seems pointless to even bother, if you send yourself a message you have already seen it.

Comment (lapulpeta) [reply Nostrid.Core/Data/DmService.cs:102]: Yes, I believe it doesn't make sense to notify self messages

**Comment by lapulpeta** [comment]
Closing for now

**Pull request closed without merging**